torch/distributions/distribution.py

```diff
@@ -72,8 +72,9 @@ def sample(self, sample_shape=torch.Size()):
         Generates a sample_shape shaped sample or sample_shape shaped batch of
         samples if the distribution parameters are batched.
         """
-        z = self.rsample(sample_shape)
-        return z.detach() if hasattr(z, 'detach') else z
+        with torch.no_grad():
+            x = self.rsample(sample_shape)
+        return x.detach() if hasattr(x, 'detach') else x
```

Comment:
I tried to remove this `.detach()` but some `.sample()`s still ended up requiring grad.

Reply:
Do you know which distributions had this problem? `x` should never require grad in here

Reply:
This fails for at least `Dirichlet` and `Beta`:
```py
param = Variable(torch.ones(2), requires_grad=True)
dist = torch.distributions.Dirichlet(param)
with torch.no_grad():
    dist.rsample().requires_grad
# True
```

Reply:
Looks like Python functions don't respect `GradMode`. Will send a fix today

Reply:
See #5184

Reply:
Thanks, that fixed it!